Formula/efm-langserver.rb

```diff
@@ -0,0 +1,19 @@
+class EfmLangserver < Formula
+  desc "General purpose Language Server"
+  homepage "https://github.com/mattn/efm-langserver"
+  url "https://github.com/mattn/efm-langserver.git",
+    tag:      "v0.0.25",
+    revision: "fe9ae6a689918107a65d4f992efc59af3c99f83f"
+  license "MIT"
+  head "https://github.com/mattn/efm-langserver.git"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", *std_go_args
+  end
+
+  test do
+    system "#{bin}/efm-langserver", "-h"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
This tool is an adapter between linters/formatters and LSP clients. As such it cannot really do much by itself.
I've updated the test to create a configuration file and then start the tool, asking it to dump the configuration back out.
I do some checks to make sure the configuration is correctly read.